cmake/compiler/gcc/target_optimizations.cmake

```diff
@@ -0,0 +1,19 @@
+# See root CMakeLists.txt for description and expectations of this macro
+macro(toolchain_cc_optimizations)
+
+  # We need to set an optimization level.
+  # Default to -Os
+  # unless CONFIG_NO_OPTIMIZATIONS is set, then it is -O0
+  # or unless CONFIG_DEBUG is set, then it is -Og
+  #
+  # also, some toolchain's break with -Os, and some toolchain's break
+  # with -Og so allow them to override what flag to use
+  #
+  # Finally, the user can use Kconfig to add compiler options that will
+  # come after these options and override them
```

Comment:
AFAICT all of the documentation above here applies to the toplevel, not to the toolchain.

Reply:
Refactored the documentation.